File: Cap 10/Topico 10.8/Recursividade Simples/recsimp.swift
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.8
 * 
 * Programa ..: FUNCAO COM RECURSIVDADE SIMPLES
 */

import Foundation

func fatorial(_ N: Int) -> UInt64 {
    if N == 0 {
        return 1
    } else {
        return UInt64(N) * fatorial(N - 1)
    }
}

var LIMITE: Int = 0

print("CALCULO DE FATORIAL")
print("RECURSIVIDADE SIMPLES\n")

print("Qual fatorial (1-20): ", terminator: "")
if let inputLIMITE = readLine(), let limite = Int(inputLIMITE) {
    LIMITE = limite
}

print("Fatorial = \(fatorial(LIMITE))")

File: Cap 10/Topico 10.8/Recursividade Simples/recsimp.jl
#=
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.8
 * 
 * Programa ..: FUNCAO COM RECURSIVDADE SIMPLES
=#

function FATORIAL(N)
  if N == 0
    return 1
  else
    return N * FATORIAL(N - 1)
  end
end

LIMITE = 0

println("CALCULO DE FATORIAL")
println("RECURSIVIDADE SIMPLES")
println()

print("Qual fatorial (1-20): ")
LIMITE = parse(Int64, readline())

println("Fatorial = ", FATORIAL(LIMITE))
